src/main/java/com/conveyal/gtfs/loader/JdbcTableWriter.java

```diff
@@ -1487,35 +1488,23 @@ private void updateReferencingTables(
             String refTableName = String.join(".", namespace, referencingTable.name);
             for (Field field : referencingTable.editorFields()) {
                 if (field.isForeignReference() && field.referenceTable.name.equals(table.name)) {
+                    String keyColumn = (table.name.equals(Table.PATTERNS.name)) ? "pattern_id" : "route_id";
                     if (
                         Table.TRIPS.name.equals(referencingTable.name) &&
                         sqlMethod.equals(SqlMethod.DELETE) &&
-                        table.name.equals(Table.PATTERNS.name)
+                        (table.name.equals(Table.PATTERNS.name) || table.name.equals(Table.ROUTES.name))
+                    ) {
+                        deleteStopTimesAndFrequencies(namespace, keyValue, keyColumn, table.name);
+                        deleteShapes(namespace, keyValue, keyColumn, table.name);
```

Comment:
I think you can get rid of the `namespace` variable, there is a field `tablePrefix` that carries the namespace (it's actually `<namespace>.` IIRC)